src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -162,6 +167,8 @@ public static class ModelBindingHelper
 
             if (await modelBinder.BindModelAsync(modelBindingContext))
             {
+                var modelValidationContext = new ModelValidationContext(modelBindingContext, modelMetadata);
+                objectModelValidator.Validate(modelValidationContext, prefix);
```

Comment:
A property like ```RootPrefix``` does not make a lot of sense inside the ```ModelValidatorContext``` primarily because its not needed at every level. I feel the usability both for the consumer of this api as well as people implementing custom ```IOjectValidator``` is better currently. Lets talk if you do not agree.